libraries/stdlib/src/kotlin/random/Random.kt

```diff
@@ -267,9 +267,13 @@ public abstract class Random {
      *
      * @sample samples.random.Randoms.defaultRandom
      */
-    companion object Default : Random() {
+    companion object Default : Random(), Serializable {
+        private const val serialVersionUID = 0L
 
-        private val defaultRandom: Random = defaultPlatformRandom()
+        private val defaultRandom: Random
+            get() = defaultPlatformRandom()
```

Comment:
We definitely do not want to evaluate this again on each call to Random.Default's methods. Please revert to storing the implementation here and resort to serialization surrogate approach.

Reply:
In JVM, this method will be 100% inlined.

Reply:
I mean, method `kotlin/random/Random$Companion.getDefaultRandom()Lkotlin/random/Random;` will be replaced with direct access to `??????ThreadLocalRandom.INSTANCE : L???;`

Reply:
@ilya-g of course, I can discard this change, but it shouldn't make any overhead on all the platforms Kotlin supports.

Reply:
Better not to speculate about platform optimization capabilities, especially considering that this code can run on other VMs such as Android runtime, when we can write it in a way that make it sure happen without any optimization.
In other Kotlin platforms, the changes caused by this one are also totally unnecessary.